File: trunk/verminmud/src/org/vermin/wicca/web/State.js

var State = {
  
  initialize: function() {
    this.sp = new FractionModel(50, 100);
    this.sp.setPreText("SP: ");
    this.spgauge = new Gauge($('spbar'), $('sptext'), this.sp);
    
    this.hp = new FractionModel(50,100);
    this.hp.setPreText("HP: ");
    this.hpgauge = new Gauge($('hpbar'), $('hptext'), this.hp);
  },
  
  update: function(hp, maxhp, sp, maxsp, exp) {
    this.sp.setMax(parseInt(new String(maxsp)));
    this.sp.setCurrent(parseInt(new String(sp)));
    this.hp.setMax(parseInt(new String(maxhp)));
    this.hp.setCurrent(parseInt(new String(hp)));
  }
}

wicca[3] = [State, null, "update"];

File: trunk/verminmud/src/org/vermin/wicca/web/State.java
package org.vermin.wicca.web;

import org.vermin.mudlib.Player;

public class State extends AbstractWebComponent implements org.vermin.wicca.remote.State {

	
	public State(WebClientOutput output) {
		super(output);
	}

	public void update(Player who) {
		StringBuilder sb = new StringBuilder();
		sb.append("{'hp':"); sb.append(who.getHp());
		sb.append(",'maxhp':"); sb.append(who.getMaxHp());
		sb.append(",'sp':"); sb.append(who.getSp());
		sb.append(",'maxsp':"); sb.append(who.getMaxSp());
		sb.append(",'exp':"); sb.append(who.getExperience());
		sb.append("}");
		
		write((short) 3, (byte) 1, Integer.toString(who.getHp()), 
				Integer.toString(who.getMaxHp()), Integer.toString(who.getSp()),
				Integer.toString(who.getMaxSp()), Long.toString(who.getExperience())); 

	}

}
